Library/Homebrew/formula_installer.rb

```diff
@@ -449,12 +449,12 @@ def expand_requirements
           Requirement.prune
         elsif req.build? && use_default_formula && req_dependency&.installed?
           Requirement.prune
+        elsif req.satisfied?
```

Comment:
I think moving this up will mess up the use of default formula for e.g. `:python` using the python formula

Reply:
Agreed.

Reply:
To be more explicit: yes, it definitely will break this.